src/elf/note.rs

```diff
@@ -205,8 +205,10 @@ if_alloc! {
             };
             debug!("{:?} - {:#x}", header, *offset);
             // -1 because includes \0 terminator
-            let name = bytes.gread_with::<&'a str>(offset, ctx::StrCtx::Length(header.n_namesz - 1))?;
-            *offset += 1;
+            let name = bytes.gread_with::<&'a str>(offset, ctx::StrCtx::Length(header.n_namesz.saturating_sub(1)))?;
+            if (header.n_namesz > 0) {
+                *offset += 1;
```

Comment:
yup i think this is the right thing to do, the offset is used in subsequent notes, via NoteIterator